main/src/main/java/com/expedia/metrics/util/Encoder.java

```diff
@@ -0,0 +1,14 @@
+package com.expedia.metrics.util;
+
+public class Encoder {
+    private static final char[] HEX_DIGITS = {'0','1','2','3','4','5','6','7','8','9','a','b','c','d','e','f'};
+
+    public static String encodeHexString(byte[] bytes) {
+        final StringBuilder builder = new StringBuilder(bytes.length*2 + 1);
```

Comment:
Pass in the StringBuilder instead of constructing one. All the places that need to use this method are appending to a StringBuilder anyway so it would be good to avoid the extra object.